hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/NonBlockingSocketReader.java

```diff
@@ -245,6 +245,20 @@ public String toString() {
         return connection + ".socketReader";
     }
 
+    @Override
```

Comment:
we can get rid of the reschedule completely. When you create the new selectionkey with the same interest set as the old key, then on the next selector.select you will be immediately notified since the underlying conditions to trigger the key in the first place, are still there. If data was available for reading, it will still be available for reading. If space is available for writing, it still is available.